### ISSUE TITLE
etnaviv/mesa/buildroot versions

### ISSUE THREAD

**Issue by npaladin2000** [retrogamehandheld/pcsx4all_openGL#1]
Little conversation i had with glebm in the channel, shed some light n what we're facing:

glebm Today at 1:02 PM
etnaviv has been merged into mesa mainline
we're just on a really old buildroot

npaladin2000 Today at 1:03 PM
So our mesa doesn't have it?
 
glebm Today at 1:04 PM
our mesa is from 2014 but we also have an ancient etna_viv
if we upgrade mesa we won't need etna_viv -- it's part of mesa now and maintained in mesa
etnaviv driver is also in mainline linux kernel now
but our kernel is too old

npaladin2000 Today at 1:05 PM
That means updating buildroot. That's a major thing..and it means using the new PCSX4ALL would require a firmware update
 
glebm Today at 1:05 PM
yes, i think step 1 would need to be to rebase buildroot and kernel customization on top of the latest versions
our buildroot is of 2014 vintage

npaladin2000 Today at 1:07 PM
Well, that's kind of a big hang-up. We'll have to wait on a new kernel/firmware before we can get started
 
glebm Today at 1:07 PM
You could probably get something work with the current drivers and kernel
but IMO this effort would be better spent updating the kernel and buildroot
mainline driver development is pretty active
https://lists.freedesktop.org/archives/dri-devel/2019-August/231410.html
 
npaladin2000 Today at 1:08 PM
We'll have to coordinate somehow
 
glebm Today at 1:08 PM
probably lots of performance benefits for switching

**Comment by jdefields** [comment]
Podulator has already built a mucher newer buildroot from opendingux.  We haven't messed with it further yet though.

Could be something I look at soon, now that initial images are building and I have a better idea of the overall system.

**Comment by npaladin2000** [comment]
Well, if we've already got the newer buildroot that's a big hurdle. That gives us current mesa, with built in etnaviv support. We don't have to figure out how to incorporate that.

**Comment by jdefields** [comment]
https://github.com/podulator/buildroot/commits/opendingux

**Comment by jdefields** [comment]
we're on an older kernel though... don't know if that's gonna be a showstopper.  I think @tonyjih was contemplating upgrading the kernel.  This would be a good reason to do so.

**Comment by npaladin2000** [comment]
I agree, if we can move up to a kernel with built in etnaviv support it will take us where we want to go.

**Comment by tonyjih** [comment]
I'm preparing to move jz-4.9 soon
